File: backend/app/schemas/replay.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel


class ReplayFrameResponse(BaseModel):
    timestamp: datetime
    price: Decimal
    spread_bps: Decimal
    depth_payload: dict | None
    regime_payload: dict | None
    ai_annotation: dict | None


class ReplaySessionResponse(BaseModel):
    id: UUID
    trade_id: UUID
    start_at: datetime
    end_at: datetime
    timeline_resolution_ms: int
    summary: dict | None
    frames: list[ReplayFrameResponse]

File: frontend/src/lib/api/replay.ts
import { api } from '@/lib/api/client'

export interface ReplayFrame {
  timestamp: string
  price: string
  spread_bps: string
  depth_payload: Record<string, number> | null
  regime_payload: Record<string, unknown> | null
  ai_annotation: { text: string; severity: string } | null
}

export interface ReplaySession {
  id: string
  trade_id: string
  start_at: string
  end_at: string
  timeline_resolution_ms: number
  summary: Record<string, unknown> | null
  frames: ReplayFrame[]
}

export async function createReplay(tradeId: string): Promise<ReplaySession> {
  return api.post(`/api/v1/replay/trades/${tradeId}`)
}
